internal/services/formatters/default_engine_formatter.go

```diff
@@ -51,27 +52,18 @@ func (f *DefaultFormatter) StartAnalysis(src string) {
 		logger.LogDebugWithLevel(messages.MsgDebugToolIgnored + tools.HorusecEngine.ToString())
 		return
 	}
-	f.svc.SetAnalysisError(f.execEngineAndParseResults(src), tools.HorusecEngine, "", src)
+	f.svc.SetAnalysisError(f.execEngineAndParseResults(), tools.HorusecEngine, "", src)
 	f.svc.LogDebugWithReplace(messages.MsgDebugToolFinishAnalysis, tools.HorusecEngine, f.language)
 }
 
-func (f *DefaultFormatter) execEngineAndParseResults(src string) error {
+func (f *DefaultFormatter) execEngineAndParseResults() error {
 	f.svc.LogDebugWithReplace(messages.MsgDebugToolStartAnalysis, tools.HorusecEngine, f.language)
 
-	findings, err := f.execEngineAnalysis(src)
+	rules := append(f.manager.GetAllRules(), f.svc.GetCustomRulesByLanguage(f.language)...)
+	findings, err := f.engine.Run(context.Background(), f.svc.GetConfigProjectPath(), rules...)
```

Comment:
We shouldn't join `f.svc.GetConfigProjectPath()` with the `src` param of `StartAnalysis` if not empty? Scenarios where we use custom work dir this may not work properly.